src/github/repository.ts

```diff
@@ -3,15 +3,30 @@ import { emitWebhookProcessedMetrics } from "utils/webhook-utils";
 export const deleteRepository = async (context, jiraClient): Promise<void> => {
 	context.log(`Deleting dev info for repo ${context.payload.repository?.id}`);
 
-	const jiraResponse = await jiraClient.devinfo.repository.delete(
+	const deleteDevinfoPromise = jiraClient.devinfo.repository.delete(
 		context.payload.repository?.id
 	);
+
+	const deleteDeploymentsPromise = jiraClient.deployment.delete(
+		context.payload.repository?.id
+	);
+
+	const deleteBuildsPromise = jiraClient.workflow.delete(
+		context.payload.repository?.id
+	);
+
+	await Promise.all([
+		deleteDevinfoPromise,
+		deleteBuildsPromise,
+		deleteDeploymentsPromise
+	]);
+
 	const { webhookReceived, name, log } = context;
 
 	webhookReceived && emitWebhookProcessedMetrics(
 		webhookReceived,
 		name,
 		log,
-		jiraResponse?.status
+		202
```

Comment:
We can just return a `202` here, because the Jira client will throw an error when it gets a non-successful response, and then this code isn't being executed anyway.